assets/activestate.yaml.tpl

```
@@ -1,5 +1,5 @@
 project: {{.Project}}
-{{if ne .LanguageName ""}}
+{{if .IncludeLanguageField }}
```

```suggestion
{{if ne .Language "" }}
```

I don't think we need a special field for this.

---

I think we do because in both cases (`activate` and `init`) that language field is not the empty string. We have to pass it for `activate` so the `activationMessage` script has a language and `init` for the `push` case. 

We could revert the if statement in the template to what you have here and then only populate the language field in the Go code if `IncludeLanguageField` is true. I'm not sure if it's more clear to have two conditionals rather than the one we have now.

---

I'd prefer for the calling logic not to have to worry about this detail. I think we can just make the conditional `{{if and (ne .Language "") (eq .CommitID "") }}`

---

Makes sense. The conditional works with a minor change to the project file creation code.